cpp/unittests/gsdkTests.cpp

```diff
@@ -172,6 +172,64 @@ namespace Microsoft
+                {
+                    GSDKInternal::testConfiguration = std::make_unique<TestConfig>("heartbeatEndpoint", "serverId", "logFolder", "sharedContentFolder");
+                    GSDK::start();
+
+                    MaintenanceSchedule schedule;
+                    GSDK::registerMaintenanceV2Callback([&schedule](MaintenanceSchedule maintenanceSchedule) -> void
+                        {
+                            schedule = maintenanceSchedule;
+                        });
+
+                    std::string responseJson =
+                        R"({
+                                "operation":"Active",
+                                "sessionConfig":
+                                {
+                                    "sessionId":"eca7e870-da2e-45f9-bb66-30d89064313a",
+                                    "sessionCookie":"OreoCookie"
+                                },
+                                "maintenanceSchedule": 
+                                {
+                                    "DocumentIncarnation": "IncarnationID",
+                                    "Events": 
+                                    [
+                                        {
+                                            "EventId": "eventID",
+                                            "EventType": "Reboot",
+                                            "ResourceType": "VirtualMachine",
+                                            "Resources": 
+                                            [
+                                                "resourceName"
+                                            ],
+                                            "EventStatus": "Scheduled",
+                                            "NotBefore": "2018-04-12T16:58:30.1458776Z",
+                                            "Description": "eventDescription",
+                                            "EventSource": "Platform",
+                                            "DurationInSeconds": 3600
+                                        }
+                                    ]
+                                },
+                                "nextHeartbeatIntervalMs":30000
+                        }")";
+                    GSDKInternal::m_instance->decodeHeartbeatResponse(responseJson);
+
+                    Assert::AreEqual("IncarnationID", schedule.m_documentIncarnation.c_str(), L"Verify maintenance V2 callback with correct document incarnation was called.");
+                    Assert::AreEqual(size_t{ 1 }, schedule.m_events.size(), L"Verify maintenance V2 callback with correct event list size was called.");
+                    Assert::AreEqual("eventID", schedule.m_events[0].m_eventId.c_str(), L"Verify maintenance V2 callback with correct event id was called.");
+                    Assert::AreEqual("Reboot", schedule.m_events[0].m_eventType.c_str(), L"Verify maintenance V2 callback with correct event type was called.");
```

Comment:
Not exactly sure how it would work, but did you consider also asCString() instead of c_str()? I see it present in this file so maybe it was there for a reason?. If you ve tested this and it works please disregard. I always am skeptical of comparing char* in c++ with '==' or sth.

Reply:
asCString is a Json::Value function that also produces a char*. This unittest framework AreEqual has specific comparers for types including char*